### PR TITLE
Update main.yml

### PR DESCRIPTION
**Pull request by Muxianesty**

### CODE REVIEW

**Comment by Muxianesty** [comment]
Pull Request is finished. Should cover issue #35.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -33,16 +28,21 @@ jobs:
...
...RCTLLHDToLLVM" "CIRCTLoopScheduleToCalyx" "CIRCTMooreToCore" "CIRCTPipelineToHW" "CIRCTSCFToCalyx" "CIRCTSeqToSV" "CIRCTSimToSV" "CIRCTCFToHandshake" "CIRCTVerifToSMT" "CIRCTVerifToSV" "CIRCTExportFIRRTL" "CIRCTComb" "CIRCTCombTransforms" "CIRCTDebug" "CIRCTESI" "CIRCTFIRRTL" "CIRCTImportFIRFile" "CIRCTMSFT" "CIRCTMSFTTransforms" "CIRCTHW" "CIRCTLLHD" "CIRCTMoore" "CIRCTOM" "CIRCTOMEvaluator" "CIRCTSeq" "CIRCTSeqTransforms" "CIRCTSV" "CIRCTSVTransforms" "CIRCTFSM" "CIRCTFSMTransforms" "CIRCTHandshake" "CIRCTHandshakeTransforms" "CIRCTHWArith" "CIRCTVerif" "CIRCTLTL" "CIRCTEmit" "CIRCTFirtool" )/foreach(_target "LLVMSupport" "LLVMCore" "LLVMMC" "LLVMTarget" "LLVMAsmParser" "LLVMBinaryFormat" "LLVMBitReader" "LLVMBitWriter" "LLVMFrontendOpenMP" "LLVMTransformUtils" "LLVMTargetParser" "LLVMIRReader" "LLVMipo" "LLVMLinker" "LLVMPasses" "LLVMMCParser" "LLVMLineEditor" "LLVMTableGen" "LLVMCoroutines" "LLVMExecutionEngine" "LLVMObject" "LLVMOrcJIT" "LLVMJITLink" "LLVMAnalysis" "LLVMAggressiveInstCombine" "LLVMInstCombine" "LLVMScalarOpts" "LLVMVectorize" "LLVMX86CodeGen" "LLVMX86Desc" "LLVMX86Info" "LLVMX86AsmParser" "LLVMX86Disassembler")/' "${{env.CIRCT_DIR}}/${{env.MLIR_TARGETS_PATH}}"
```
Comment (ssmolov) [inline .github/workflows/main.yml:39]: split into smaller lines

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -33,16 +28,21 @@ jobs:
...
+          cd "${{github.workspace}}"
       - name: Configure Utopia HLS build
+        env:
+          CIRCT_DIR: "firtool-${{env.CIRCT_VERSION}}"
         run: |
-          cmake -S . -B "${{env.BUILD_DIR}}" -G Ninja -DCMAKE_CXX_COMPILER=g++ -DCMAKE_PREFIX_PATH="${{GITHUB_WORKSPACE}}/${{env.CIRCT_PATH}}" -DSRC_FILES="${{env.DEFAULT_KERNEL}}" -DBUILD_TESTS=ON
+          cmake -S . -B "${{env.BUILD_DIR}}" -G Ninja -DCMAKE_CXX_COMPILER=g++ -DCMAKE_PREFIX_PATH="${{github.workspace}}/${{env.DEPENDENCIES_DIR}}/${{env.CIRCT_DIR}}" -DSRC_FILES="${{github.workspace}}/examples/polynomial2/polynomial2.cpp" -DBUILD_TESTS=ON
```
Comment (ssmolov) [inline .github/workflows/main.yml:45]: split into smaller lines

File: .github/workflows/main.yml (hunk 1 above)
Comment (Muxianesty) [reply .github/workflows/main.yml]: Fixed.

File: .github/workflows/main.yml (hunk 2 above)
Comment (Muxianesty) [reply .github/workflows/main.yml]: Fixed.

**Review by ssmolov — APPROVED** [review]

**Pull request merged**